src/Altinn.App.Core/Features/Correspondence/Builder/CorrespondenceAttachmentBuilder.cs

```diff
@@ -0,0 +1,100 @@
+using Altinn.App.Core.Features.Correspondence.Models;
+
+namespace Altinn.App.Core.Features.Correspondence.Builder;
+
+/// <summary>
+/// Builder factory for creating <see cref="CorrespondenceAttachment"/> objects
+/// </summary>
+public class CorrespondenceAttachmentBuilder : CorrespondenceBuilderBase, ICorrespondenceAttachmentBuilder
+{
+    private string? _filename;
+    private string? _name;
+    private string? _sendersReference;
+    private string? _dataType;
+    private ReadOnlyMemory<byte>? _data;
+    private bool? _isEncrypted;
+    private CorrespondenceDataLocationType _dataLocationType =
+        CorrespondenceDataLocationType.ExistingCorrespondenceAttachment;
+
+    private CorrespondenceAttachmentBuilder() { }
+
+    /// <summary>
+    /// Creates a new <see cref="CorrespondenceAttachmentBuilder"/> instance
+    /// </summary>
+    public static ICorrespondenceAttachmentBuilderFilename Create() => new CorrespondenceAttachmentBuilder();
```

Comment:
If I understand the question correctly, it's because of the returned interface. We need to new-up a builder instance, but only return a specific interface that says "I now need the filename, please and thank you".